Source/WebCore/layout/integration/inline/InlineIteratorTextBox.cpp

```diff
@@ -43,6 +43,29 @@ TextBoxIterator TextBox::nextTextBox() const
     return TextBoxIterator(*this).traverseNextTextBox();
 }
 
+Vector<LayoutRect> TextBox::characterRects(unsigned rangeStart, unsigned rangeEnd) const
+{
+    auto [clampedStart, clampedEnd] = selectableRange().clamp(rangeStart, rangeEnd);
+    if (clampedStart >= clampedEnd)
+        return { };
+
+    if (auto* svgTextBox = dynamicDowncast<SVGInlineTextBox>(legacyInlineBox())) {
+        Vector<LayoutRect> rects;
+        rects.reserveInitialCapacity(clampedEnd - clampedStart);
+        for (auto index = clampedStart; index < clampedEnd; ++index)
+            rects.uncheckedAppend(svgTextBox->localSelectionRect(index, index + 1));
+        return rects;
+    }
+
+    auto lineSelectionRect = LineSelection::logicalRect(*lineBox());
+    auto selectionRect = LayoutRect { logicalLeftIgnoringInlineDirection(), lineSelectionRect.y(), logicalWidth(), lineSelectionRect.height() };
+
+    auto result = fontCascade().characterSelectionRectsForText(textRun(), selectionRect, clampedStart, clampedEnd).map([&](auto& characterRect) {
+        return snappedSelectionRect(characterRect, logicalRightIgnoringInlineDirection(), lineSelectionRect.y(), lineSelectionRect.height(), isHorizontal());
+    });
+    return result;
+}
```

Comment:
Is there a particular reason for having this on InlineIterator::TextBox (other than that similar functions are already on TextBox)?

Reply:
Indeed — the only reason is that other methods that behave similarly are already there — but since it's only called in one place, perhaps it would be cleaner to turn this into a static helper function near the call site? I'll give that a try.

Reply:
ok, I was planning on moving those methods out of the InlineIterator::TextBox to keep this interface clean (and simple).  -they don't really belong in there.